presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2141,6 +2141,27 @@ public void testJoinWithLessThanOnDatesInJoinClause()
     }
 
     @Test
+    public void testJoinWithNonDeterministicLessThan()
+    {
+        MaterializedRow actualRow = getOnlyElement(computeActual(
```

Comment:
You just do two subqueries with count(*) aggregation and then compare results in a single query. What do you think?